qa/sc_evm_test_metamask_related.py

```diff
@@ -409,19 +401,31 @@ def run_test(self):
         eoa_transfer(sc_node, evm_address, other_address, transfer_amount, static_call=True, generate_block=False)
         eoa_transfer(sc_node, evm_address, other_address, transfer_amount, call_method=CallMethod.RPC_LEGACY)
 
-        eoa_assert_native_balance(sc_node, evm_address, initial_balance - transfer_amount)
+        current_balance = int(sc_node.rpc_eth_getBalance(str(evm_address), "latest")['result'], 16)
+        total_gas_used = initial_balance - (current_balance + transfer_amount)
+
+        eoa_assert_native_balance(sc_node, evm_address, current_balance - (transfer_amount + total_gas_used))
```

Comment:
This should be `initial_balance - ...` right?

Additionally, I don't think we can test like this. By fetching the balance to calculate the fees and then verifying the balance this becomes a self-fulfilling test. An alternative would be to fetch the receipt for the transaction, get the `usedGas` and multiply it with the `gasPrice` we set in the transaction. This would work at least for legacy transactions. I don't know if we can fetch receipts yet, though.